docs/source/reference/pysupported.rst

```diff
@@ -160,10 +160,40 @@ Numba will reject any list containing objects of different types, even if
 the types are compatible (for example, ``[1, 2.5]`` is rejected as it
 contains a :class:`int` and a :class:`float`).
 
+For example, to create a list of arrays::
+
+  In [1]: from numba import njit
+
+  In [2]: import numpy as np
+
+  In [3]: @njit
+    ...: def foo(x):
+    ...:     lst = []
+    ...:     for i in range(x):
+    ...:         lst.append(np.arange(i))
+    ...:     return lst
+    ...:
+
+  In [4]: foo(4)
+  Out[4]: [array([], dtype=int64), array([0]), array([0, 1]), array([0, 1, 2])]
+
+
+List Reflection
+'''''''''''''''
+
+In nopython mode, Numba does not operate on Python objects.  ``list`` are
+compiled into an internal representation.  Any ``list`` arguments must be
+converted into this representation on the way in and their contained elements
+must be restored in the original Python objects via a process called
+*reflection*.  Reflection is required to maintain the same semantic as regular
```

Comment:
This names the process "reflection" and says it is required, but doesn't say what it actually does. Is there a FAQ entry that can be linked in here?